src/helpers/models.ts

```diff
@@ -187,13 +187,15 @@ export function delegatedStakeExists(
 export function createAllocation(
   allocationID: string,
   indexerID: string,
-  epochCreated: BigInt
+  epochCreated: BigInt,
+  subgraphDeploymentID: string
 ): void {
   if (Allocation.load(allocationID) == null) {
     let allocation = new Allocation(allocationID);
     allocation.createdAtEpoch = epochCreated;
     allocation.indexer = indexerID;
 
+    allocation.subgraphDeploymentID = subgraphDeploymentID;
```

Comment:
done

Reply:
There are a lot of questions as to how subgraphs will scale on the decentralized network. Since queries will cost GRT, subgraphs that index unnecessary data effectively makes querying the subgraph more expensive.

On another front, there's a massive refactor I'm about to push that adds subgraph entities and a couple other entities.

Reply:
Thank you for the review, @HardForksOverKnives ! Please let me know how that refactor goes, if there is anything I can help with. Are there any new entities I could take on, try to start indexing on the badgeth subgraph?

> Since queries will cost GRT, subgraphs that index unnecessary data effectively makes querying the subgraph more expensive.
So, there is a balance between indexing all that is needed and efficiency, there is a cost to over-indexing.